Cookbook.Domain/Models/UnitType.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Cookbook.Domain.Models
+{
+    public enum UnitType
+    {
+        kg = 1,
+        dkg = 2,
+        g = 3,
+
+        l = 4,
+        dl = 5,
+        cl = 6,
+        ml = 7,
+
+        piece = 8,
+        pieces = 9,
+        clove = 10,
```

Comment:
Ez a fokhagyma miatt? Nem lehet arra is piece-t hasznalni? 

Lehet jobb lenne ez valami string-kent mint enum-kent. Ez vegtelensegig bovulhet az ingredient-tol meg recept-tol fuggoen. Cup, slice, can, bottle, etc?

Reply:
de akkor hogy kezelned azt a reszt h ugyanaz legyen a string? alapbol onnan jott az enum otlete h nem akartam h legyen kg kilo kilogram kilogramm etc hanem egy kontrolalt verzio van csak

Reply:
hat ez jo kerdes, ez kicsit inkabb UX design kerdes, mint szoftver tervezes kerdes. Itt alapvetoen a nagy UX kerdes, hogy kik editalhatjak a recepteket. 
1. Maga a UX "render" meg lokalizacio, amit fent emlitettem csak akkor tervezheto (pl roviditunk kis kepernyon, kiirjuk nagyon), ha elore tudjuk a lehetseges unit-okat. Ebben az esetben egy szuk admin kor tudna recepteket edit-elni, akikkel kapcsolatban all a szoftver csapat es tudjak igazitani a unit-okat (mint feature request) az alapjan hogy mire van szukseg. Ez gyakorlatilag az az irany amit most te kovetsz, mert az enum csak compile time valtoztathato. 
Ebben az iranyban sem kell amugy ragaszkodni az enum-hoz, de developer-ek adhatnak csak hozza uj unit-ot, akkor nem lesz kg/kilo/kilogram issue, de talan karbantarthatobb a kod. 
2. Masik opcio, hogy barki hozzaadhat ujat. Ebben az esetben lenne valami UI felulet ahol te definialhatod a unit-odat, meg esetleg megadhatsz konverziot mas mar letezo unit-hoz, hogy segitsd a rendszert atvaltas soran. Ez relative nagy szabadsag az oldalon, mert akkor tenyleg irhatok befottes uveget vagy barmit, cserebe nem fog mukodni a lokalizacio, mert nem tudja a fejleszto csapat release time hogy miket kene lefordittatni. Egyes szam/tobbes szam se igazan mukodik mert nem tudom mi a tobbes szama annak a random string-nek amit beirtal
3. Egy koztes megoldas, en valoszinu ebbe az iranyba mennek ha nekem kene UX-rol dontenem (ami mar regen rossz), ez valahol a ketto kozott lenne feluton. Lenne egy szerkeszto felulet es adatbazisban tartanank a hasznalhato unit-ok listajat, de csak adott permission-nel erheto el a felulet (i.e. csak admin-oknak). Igy a lokalizaciot elbukjuk tovabbra is, de ha egy "ismert" user bazis szerkeszti csak, akkor kicsit advance-abb field-eket is hozza lehet adatni veluk (singualr, plural, rovidites, etc), es akkor a UX render resze meg mukodhet. 

Long story short, az nem a te feladatod amikor adat tipusokat valasztasz hogy "lekorlatozd" a user-t idiota duplikaciok hozzaadasaban. Az maga egy feature, ha erre szukseg van, akkor front end-en le kell fejleszteni. Az adat modellednek az a dolga hogy mas fejlesztok idiotizmusatol vedjek a kodot, nem az hogy a userek idiotizmusatol vedjek az adatot. 

Szerintem az nem egy jo practice hogy ujra kell forditani a kodot ha hozza akarunk adni egy uj unit-ot, szoval en inkabb string-et hasznalnek es az adatbazisba raknek egy unit tablat. Az mar mas kerdes hogy ki tolti fel azt a tablat, lehet siman az init script ami letrehozza a tablat, feltolti ugyanezt a listat es kesz, read only, meg lehet user is aki feltolti, es akkor jon hozza valami feature ami ebben segit. 

Mi csinaltunk Saxo-nal amugy olyant, az szerintem jo practice volt, hogy ebben az esetben van egy unit lista az init script-ben, es ha jon egy feature request hogy kene "bottle" is, akkor siman hozzaadjuk az adatbazishoz live ES update-ljuk a script-et hogy legkozelebb amikor nullarol letrehozza az adatbazist akkor mar rakja bele egybol.